controller/controller.go

```diff
@@ -65,6 +67,14 @@ func DefaultConfig() Config {
 // Controller defines the interface a controller needs to implement to provide
 // operations for groups of unit files against a fleet cluster.
 type Controller interface {
+	// GroupNeedsUpdate checks if the given group should be updated or not. To
+	// make a decision the unit content of each unit of each slice is compared
+	// using its unit hash. As soon as one unit hash differs, or a unit cannot be
+	// found, Inago assumes the whole group slice to be "dirty" and returns true
+	// having the group slices removed from the given req that are up to date,
+	// otherwise false, leaving the req as it is.
+	GroupNeedsUpdate(req Request) (Request, bool, error)
```

Comment:
:one: The description suggests I get the slices that do not need to be changed. The implementation looks like I only get the dirty ones. I also find the behavior to return the original array when nothing changed surprising. Can't we just return the dirty slices? If they are returned, they are all dirty. It's the client responsability to make a new request out of this.

:two: Why do we want this in the public interface?